src/bin/lpython.cpp

```diff
@@ -840,6 +840,106 @@ int compile_python_to_object_file(
+    if (!r.ok) {
+        print_time_report(times, time_report);
+        return 1;
+    }
+
+    // Src -> AST -> ASR
+    LCompilers::LPython::AST::ast_t* ast = r.result;
+    diagnostics.diagnostics.clear();
+    auto ast_to_asr_start = std::chrono::high_resolution_clock::now();
+    LCompilers::Result<LCompilers::ASR::TranslationUnit_t*>
+        r1 = LCompilers::LPython::python_ast_to_asr(al, lm, nullptr, *ast, diagnostics, compiler_options,
+            !(false && compiler_options.po.disable_main), "__main__", infile);
+
+    auto ast_to_asr_end = std::chrono::high_resolution_clock::now();
+    times.push_back(std::make_pair("AST to ASR", std::chrono::duration<double, std::milli>(ast_to_asr_end - ast_to_asr_start).count()));
+    std::cerr << diagnostics.render(lm, compiler_options);
+    if (!r1.ok) {
+        LCOMPILERS_ASSERT(diagnostics.has_error())
+        print_time_report(times, time_report);
+        return 2;
+    }
+    LCompilers::ASR::TranslationUnit_t* asr = r1.result;
+    if( compiler_options.po.disable_main ) {
+        int err = LCompilers::LPython::save_pyc_files(*asr, infile);
+        if( err ) {
+            return err;
+        }
+    }
+    diagnostics.diagnostics.clear();
+
+    // ASR -> LLVM
+    if (compiler_options.emit_debug_info) {
+#ifndef HAVE_RUNTIME_STACKTRACE
+        diagnostics.add(LCompilers::diag::Diagnostic(
+            "The `runtime stacktrace` is not enabled. To get the stacktraces, "
+            "re-build LPython with `-DWITH_RUNTIME_STACKTRACE=yes`",
+            LCompilers::diag::Level::Error,
+            LCompilers::diag::Stage::Semantic, {})
+        );
+        std::cerr << diagnostics.render(lm, compiler_options);
+        return 1;
+#endif
+    }
+    LCompilers::PythonCompiler fe(compiler_options);
+    LCompilers::LLVMEvaluator e(compiler_options.target);
+    auto asr_to_llvm_start = std::chrono::high_resolution_clock::now();
+    LCompilers::Result<std::unique_ptr<LCompilers::LLVMModule>>
+        res = fe.get_llvm3(*asr, pass_manager, diagnostics, infile);
+    auto asr_to_llvm_end = std::chrono::high_resolution_clock::now();
+    times.push_back(std::make_pair("ASR to LLVM", std::chrono::duration<double, std::milli>(asr_to_llvm_end - asr_to_llvm_start).count()));
+
+    std::cerr << diagnostics.render(lm, compiler_options);
+    if (!res.ok) {
+        LCOMPILERS_ASSERT(diagnostics.has_error())
+        print_time_report(times, time_report);
+        return 3;
+    }
+
+    auto llvm_start = std::chrono::high_resolution_clock::now();
+    e.add_module(std::move(res.result));
+    try {
+        e.voidfn("__module___main_____main__global_stmts");
+    } catch (LCompilers::LCompilersException&) {}
+    auto llvm_end = std::chrono::high_resolution_clock::now();
+    times.push_back(std::make_pair("LLVM to binary", std::chrono::duration<double, std::milli>(llvm_end - llvm_start).count()));
+    print_time_report(times, time_report);
```

Comment:
Let's do this only for LLVM backend.
This seems duplicated, try utilising or simplifying the function: `compile_python_to_object_file`